common/src/main/java/org/conscrypt/Conscrypt.java

```diff
@@ -370,6 +370,14 @@ public static void setBufferAllocator(SSLEngine engine, BufferAllocator bufferAl
         toConscrypt(engine).setBufferAllocator(bufferAllocator);
     }
 
+    /**
+     * Provides a {@link BufferAllocator} instance to be used by all conscrypt engines by default.
```

Comment:
I think the phrasing you have in ConscryptEngine is clearer, so I'd go with something like "Configures the default {@link BufferAllocator} to be used by all future {@link SSLEngine} instances from this provider."